packages/backend-common/src/reading/tree/ReadTreeResponseFactory.ts

```diff
@@ -24,8 +24,9 @@ import { ZipArchiveResponse } from './ZipArchiveResponse';
 type FromArchiveOptions = {
   // A binary stream of a tar archive.
   stream: Readable;
-  // If set, the root of the tree will be set to the given directory path.
-  path?: string;
+  // If unset, the files at the root of the tree will be read.
+  // subpath must not contain the name of the top level directory.
```

Comment:
> So this is single-dir-at-root is true for all of the readTree sources?

Yeah I think so. We are downloading a zip/tarball of the repository regardless of the path requested by the user. So,
`reader.readTree("https://github.com/backstage/backstage")`
and
`reader.readTree("https://github.com/backstage/backstage/tree/master/docs")`

will download the same `something-master.tar.gz` which will have the single dir at the root. It's kept in memory and not extract to filesystem.

When `response.dir()` or `response.files()` is called, it will output files and their paths relative to the subpath provided - `/` or `/docs`.

So, 

```
const response = await reader.readTree("https://github.com/backstage/backstage/tree/master/docs/conf")
const dir = await response.dir()

// ls dir ->
// defining.md
// index.md
// reading.md
// writing.md
```

```
const response = await reader.readTree("https://github.com/backstage/backstage")
const dir = await response.dir()

// ls dir ->
// .changeset
// .github
// ...
```